docs/develop/dotnet/message-passing.mdx

```diff
@@ -1,261 +1,791 @@
+
+After failing to find a handler with a matching name and type, the Worker checks for a registered dynamic stand-in handler.
+If found, the Worker uses that instead.
+
+You must opt handlers into dynamic access.
+For example, you make a Signal dynamic by adding `Dynamic = true` to the `[WorkflowSignal]` attribute.
+
+The handler must accept a string input for the name, and a `Temporalio.Converters.IRawValue[]` for the arguments.
+Use a [Workflow.PayloadConverter](https://dotnet.temporal.io/api/Temporalio.Workflows.Workflow.html#Temporalio_Workflows_Workflow_PayloadConverter)  to convert `IRawValue` objects to your required type.
+For example:
 
 ```csharp
-[WorkflowUpdate]
-public async Task<string> UpdateStatusAsync(Status status)
+[WorkflowSignal(Dynamic = true)]
+public async Task DynamicSignalAsync(string signalName, IRawValue[] args)
 {
-    this.status = status;
-    return "Status updated";
+    var input = Workflow.PayloadConverter.ToValue<DoSomethingParam>(args.Single());
+    pendingThings.Add(input);
 }
 ```
 
-### Send an Update from a Temporal Client {#send-update-from-client}
+This sample creates a dynamic Signal.
+When an unregistered or unrecognized Signal arrives with a matching signature, dynamic assignment uses this handler to manage the Signal.
+It is responsible for transforming the sequence contents into usable data in a form that the method's logic can process and act on.
+
+When registered, any Signals sent to the Workflow without a defined handler will be delivered to the dynamic `WorkflowSignal`.
+Note that you can only register one each of a dynamic Signal, Query, or Update per Workflow Execution.
+Dynamic handlers can be implemented in both regular and dynamic Workflow implementations.
+
+
+<div style={{backgroundColor: '#ffff00',padding: '0px 15px',borderRadius: '5px',border: '1px solid #cccccc',display: 'inline-block'}}>**.NET source did not include dynamic Workflow coverage.**</div>
+
+### Set a dynamic Workflow {#set-a-dynamic-workflow}
+
+A dynamic Workflow refers to a special stand-in Workflow Definition.
+It's used when an unknown Workflow Execution request arrives.
+
+Consider the "MyUnknownWorkflow" example described earlier.
+The Worker may find there's no registered Workflow Definitions of that name or type.
+After failing to find a Workflow Definition with a matching type, the Worker looks for a dynamic stand-in.
+If found, it invokes that instead.
+
+To participate, your Workflow must opt into dynamic access.
+Adding `implements DynamicWorkflow` to the Workflow class declaration makes the Workflow Definition eligible to participate in dynamic invocation.
+You must register the Workflow with the [Worker](https://python.temporal.io/temporalio.worker.html) before it can be invoked.
+
+```csharp
+NEEDS EXAMPLE
+```
+
+<div style={{backgroundColor: '#ffff00',padding: '0px 15px',borderRadius: '5px',border: '1px solid #cccccc',display: 'inline-block'}}>**Needs .NET.**</div>
```

Comment:
Dynamic workflows and activities are with workflow and activity docs instead of with query, signal, update docs